sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -704,16 +811,71 @@ public HttpPipeline getHttpPipeline() {
             marker -> StorageImplUtils.applyOptionalTimeout(this.client.getMessages()
                 .dequeueWithResponseAsync(queueName, maxMessages, visibilityTimeoutInSeconds,
                     null, null, context), timeout)
-                .map(response -> new PagedResponseBase<>(response.getRequest(),
-                    response.getStatusCode(),
-                    response.getHeaders(),
-                    response.getValue(),
-                    null,
-                    response.getDeserializedHeaders()));
+                .flatMap(this::transformMessagesDequeueResponse);
 
         return new PagedFlux<>(() -> retriever.apply(null), retriever);
     }
 
+    private Mono<PagedResponseBase<MessagesDequeueHeaders, QueueMessageItem>> transformMessagesDequeueResponse(
+        MessagesDequeueResponse response) {
+        List<QueueMessageItemInternal> queueMessageInternalItems = response.getValue();
+        List<QueueMessageItem> queueMessageItems = Collections.emptyList();
+        Mono<Void> mono = Mono.empty();
+        if (queueMessageInternalItems != null) {
```

Comment:
If we inverted this to `queueMessageInternalItems == null` -> return empty paged response, I think we could streamline the conversion logic to use a fully reactive structure.

```java
return Flux.fromIterable(queueMessageInternalItems)
    .map(/* conversion logic */)
    .collectList()
    .map(/* create paged response */);
```